### DISCUSSION TITLE
Execute script

### DISCUSSION THREAD

**Discussion by rybkam** [zariiii9003/pycanape#3] [General]
Hi.
Is there an example of a script call?
I try it like this. 
![image](https://github.com/zariiii9003/pycanape/assets/135838086/db2ed354-5608-4085-bd35-ffeb7f82e4e8)

But I get an error.
![image](https://github.com/zariiii9003/pycanape/assets/135838086/a6005492-e755-4b6e-83a6-2a0c96f037f5)

In CANape I have this log.
![image](https://github.com/zariiii9003/pycanape/assets/135838086/3350d0bf-b0ca-4eb5-a64b-76f308f45137)

**Comment by zariiii9003** [comment]
I created a minimal script for testing:
![image](https://github.com/zariiii9003/pycanape/assets/52598363/3f933f89-da86-4dd2-af5a-0b4db0dbbe47)

And this is my python code:
```python3
import time

from pycanape import CANape
from pycanape.cnp_api.cnp_constants import TScriptStatus

CANAPE_PROJECT_PATH = r"C:\path\to\my\project"
DEVICE_NAME = "DUT"

if __name__ == '__main__':
    canape = CANape(
        project_path=CANAPE_PROJECT_PATH,
        modal_mode=True,
    )
    print(canape.get_application_version())
    print(canape.get_dll_version())

    dut = canape.get_module_by_name(DEVICE_NAME)
    script = dut.execute_script_ex(True, "myScript.cns")
    script.start_script()
    while script.get_script_state() is not TScriptStatus.eTScrFinishedReturn:
        time.sleep(0.01)
    print(script.get_script_result_string())
    canape.exit(False)
```

This prints
```text
AppVersion(main_version=20, sub_version=0, service_pack=30, app_name='CANape')
DllVersion(dll_main_version=2, dll_sub_version=5, dll_release=0, os_version='Windows95/WindowsNT', os_release=0)
Great Success
```

There's actually a bug in the latest release, so it fails on `get_script_result_string()`, but that is fixed on the master branch.


Edit: Can you show the output of `print(scr.script_handle)` after `execute_script_ex`?

**Reply by rybkam** [reply]
I used your code. I added `print(script.script_handle)`. The effect is unchanged. I am using release 0.6.0.
Output:

```
AppVersion(main_version=20, sub_version=0, service_pack=22, app_name='CANape')
DllVersion(dll_main_version=2, dll_sub_version=5, dll_release=0, os_version='Windows95/WindowsNT', os_release=0)
c_ulong(280206805)
Traceback (most recent call last):
  File "D:\CANape_test_runner\0901_049_SWIT\tester.py", line 20, in <module>
    script.start_script()
  File "D:\rybkamD\python3_10\lib\site-packages\pycanape\script.py", line 54, in start_script
    cnp_prototype.Asap3StartScript(
  File "D:\rybkamD\python3_10\lib\site-packages\pycanape\utils.py", line 28, in wrapper
    return func(*args, **kwargs)
OSError: exception: access violation reading 0x0000000000000001

```

**Reply by zariiii9003** [reply]
Did you also use my minimal CASL script?

**Reply by rybkam** [reply]
Yes, I also use your CASL script.

**Reply by zariiii9003** [reply]
I just noticed, that the argument list is not correct:
![image](https://github.com/zariiii9003/pycanape/assets/52598363/019e7385-10ce-4a68-9d98-b43e4b472e33)

**Reply by zariiii9003** [reply]
I just uploaded a fix, could you try again with the master branch?

**Reply by rybkam** [reply]
The fix works. Thanks a lot.
```
$ python tester.py
AppVersion(main_version=20, sub_version=0, service_pack=22, app_name='CANape')
DllVersion(dll_main_version=2, dll_sub_version=5, dll_release=0, os_version='Windows95/WindowsNT', os_release=0)
c_ulong(575821872)
=== TEST PASS ===
```

**Reply by zariiii9003** [reply]
Great, I'll create a bugfix release then
